Modules/pyexpat.c

```diff
@@ -1816,25 +1857,39 @@ MODULE_INITFUNC(void)
 
     if (PyModule_AddStringConstant(errors_module, "__doc__",
                                    "Constants used to describe "
-                                   "error conditions.") < 0)
-        return NULL;
+                                   "error conditions.") < 0) {
+        goto error;
+    }
 
-    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0)
-        return NULL;
-    if (PyModule_AddObject(errors_module, "messages", rev_codes_dict) < 0)
-        return NULL;
+    if (PyModule_AddObject(errors_module, "codes", codes_dict) < 0) {
+        goto error;
+    }
+    if (PyModule_AddObject(errors_module, "messages", rev_codes_dict) < 0) {
+        goto error;
+    }
 
 #undef MYCONST
 
-#define MYCONST(c) PyModule_AddIntConstant(m, #c, c)
+#define MYCONST(c) do {                                 \
+        if (PyModule_AddIntConstant(mod, #c, c) < 0) {  \
+            goto error;                                  \
+        }                                               \
+    } while(0)
     MYCONST(XML_PARAM_ENTITY_PARSING_NEVER);
```

Comment:
You may put do on a seprated line and add an empty line for readbility:

```suggestion
#define MYCONST(c) \
    do { \
        if (PyModule_AddIntConstant(mod, #c, c) < 0) { \
            goto error; \
        } \
    } while(0)
    
    MYCONST(XML_PARAM_ENTITY_PARSING_NEVER);
```

Same remark for the redefined MYCONST below.